### ISSUE TITLE
Documentation string representation discrepancy

### ISSUE THREAD

**Issue by Alexander-Shukaev** [Wilfred/helpful#197]
Given `counsel`:

```emacs-lisp
(defcustom counsel-find-file-ignore-regexp nil
  "A regexp of files to ignore while in `counsel-find-file'.
These files are un-ignored if `ivy-text' matches them.  The
common way to show all files is to start `ivy-text' with a dot.

Example value: \"\\(?:\\`[#.]\\)\\|\\(?:[#~]\\'\\)\".")
```

Output of `helpful`:

```
counsel-find-file-ignore-regexp is a variable defined in counsel.el.

Value
\(?:\`[#.]\)\|\(?:[#~]\'\)

View as literal Set Customize

Documentation
A regexp of files to ignore while in counsel-find-file.

These files are un-ignored if ivy-text matches them.  The common way to show all files is to start ivy-text with a dot.

Example value: "\(?:M-x #.\)\|\(?:[#~]\\)".
```

Output of `help`:

```
counsel-find-file-ignore-regexp is a variable defined in ‘counsel.el’.
Its value is "\\(?:\\`[#.]\\)\\|\\(?:[#~]\\'\\)"
Original value was nil

Documentation:
A regexp of files to ignore while in ‘counsel-find-file’.
These files are un-ignored if ‘ivy-text’ matches them.  The
common way to show all files is to start ‘ivy-text’ with a dot.

Example value: "\(?:\‘[#.]\)\|\(?:[#~]\’\)".
```

I observe 3 sub-issues here:

1.  ``"\\(?:\\`[#.]\\)\\|\\(?:[#~]\\'\\)"`` is definitely better than ``\(?:\`[#.]\)\|\(?:[#~]\'\)`` because it can also be copied/pasted directly into Emacs Lisp and give clear verbatim representation of the value.
2.  `Original value was nil` is very useful.  Is it supported by `helpful` at all?
3.  Notice how ``\"\\(?:\\`[#.]\\)\\|\\(?:[#~]\\'\\)\"`` from documentation string is misrendered by `helpful` as ``"\(?:M-x #.\)\|\(?:[#~]\\)"``, and while `help` does better job here, ``"\(?:\‘[#.]\)\|\(?:[#~]\’\)"``, it is still not ideal as `` ` `` becomes `‘` and `'` becomes `’`.

**Comment by Wilfred** [comment]
OK, I've fixed the issues you've mentioned with string literal rendering :)

Would you mind opening a separate issue about showing the default value for custom variables? That's a feature helpful doesn't have yet.

> it is still not ideal as ` becomes ‘ and ' becomes ’.

Yeah, there was a bunch of controversy on emacs-devel about this. Some people really like curly quotes in their docstrings, whereas others preferred a more literal interpretation of quotes. I like my `` ` `` to not get converted. You can get this behaviour in normal Help buffers by `(setq text-quoting-style 'grave)`.

**Comment by Alexander-Shukaev** [comment]
Thanks!  I will create separate issues where we can continue discussion.

**Comment by Alexander-Shukaev** [comment]
Probably one more point that I wanted to make about sub-issue 3 is the following.  Is it somehow possible to render everything between `\"` and `\"` verbatim?  Clearly, when a someone writes

```
Example value: \"\\(?:\\`[#.]\\)\\|\\(?:[#~]\\'\\)\"
```

the desired or ideal rendering would be

```
Example value: "\\(?:\\`[#.]\\)\\|\\(?:[#~]\\'\\)"
```

At the moment, this can only be achieved via double escaping (`\\\\`).

**Comment by Wilfred** [comment]
Docstring rendering escaping should be fixed as of 0a83a7b.

**Comment by notetiene** [comment]
Sorry for being late.

The docstring escaping is breaking docstrings semantic.  The only way for that feature to be useful is if `help` start using the escape mechanism of `helpful`.  The double escaping method has been the standard method since docstrings existed and it seems to be used everywhere in emacs.  I can count at least 20 cases when it's used only in function docstrings.  The most problematic I've seen is for `interactive`.  Using helpful, it gives:

```
For example, write
 (defun foo (arg buf) "Doc string" (interactive "P\\nbbuffer: ") .... )
 to make ARG be the raw prefix argument, and set BUF to an existing buffer,
 when foo is called as a command.
```

where:
```emacs-lisp
(defun foo (arg buf) "Doc string" (interactive "P\\nbbuffer: "))
```

is not a valid function when called interactively.

```emacs-lisp
(defun foo (arg buf) "Doc string" (interactive "P\nbbuffer: "))
```
is the intended form as seen by the `describe-function` command.

I was going to fill a bug, but hopefully I saw that issue.  Thanks for that project.
